content/influxdb/cloud-dedicated/reference/sql/functions/time-and-date.md

```diff
@@ -360,6 +362,279 @@ GROUP BY _time, room
+| time                      | room        |           avg_temp |
+| :------------------------ | :---------- | -----------------: |
+| 2022-01-01T00:00:00-08:00 | Kitchen     |  22.61666666666667 |
+| 2022-01-01T12:00:00-08:00 | Kitchen     |               22.7 |
+| 2022-01-01T00:00:00-08:00 | Living Room | 22.166666666666668 |
+| 2022-01-01T12:00:00-08:00 | Living Room |               22.2 |
+
+{{% /influxdb/custom-timestamps %}}
+
+{{% /expand %}}
+{{< /expand-wrapper >}}
+
+## date_bin_wallclock_gapfill
+
+Calculates time intervals using the timezone of a specified time value and
+returns the start of the interval nearest to the specified timestamp.
+If no rows exist in a time interval, a new row is inserted with a `time` value
+set to the interval start time, all columns in the `GROUP BY` clause populated,
+and null values in aggregate columns.
+
+Use `date_bin_wallclock_gapfill` with [`interpolate`](/influxdb/cloud-dedicated/reference/sql/functions/misc/#interpolate)
+or [`locf`](/influxdb/cloud-dedicated/reference/sql/functions/misc/#locf) to
+[fill gaps in data]()
+at specified time intervals in a specified time zone.
+
+### Time zone shifts
+
+Many regions use time zone shifts (such as Daylight Saving Time (DST) and British
+Summer Time (BST)). If a wall clock time bin starts at a time that does not
+exist in the specified time zone, the timestamp is adjusted to the time that is
+the same offset from the start of the day in that time zone.
+
+If a wall clock time represents an ambiguous time in the region then the
+behavior depends on the size of the specified interval. If the interval is
+larger than the difference between the two possible timestamps, then the earlier
+timestamp is used. Otherwise, the function uses the timestamp that matches the
+UTC offset of the input timestamp.
+
+```sql
+date_bin_wallclock_gapfill(interval, expression[, origin_timestamp])
+```
+
+{{% note %}}
+`date_bin_wallclock_gapfill` requires [time bounds](/influxdb/cloud-dedicated/query-data/sql/basic-query/#query-data-within-time-boundaries)
+in the `WHERE` clause.
+{{% /note %}}
+
+##### Arguments:
+
+- **interval**: Bin interval.
+- **expression**: Time expression to operate on.
+  Can be a constant, column, or function.
+  The output timestamp uses the time zone this time expression.
+- **origin_timestamp**: Starting point used to determine bin boundaries.
+  This must be a "wall clock" timestamp (no time zone).
+  _Default is the Unix epoch._
+
+The following intervals are supported:
+
+- nanoseconds
+- microseconds
+- milliseconds
+- seconds
+- minutes
+- hours
+- days
+- weeks
+
+> [!Note]
+> `date_bin_wallclock_gapfill` does _not_ support month-, year-, or century-based intervals.
+
+##### Related functions
+
+[interpolate](/influxdb/cloud-dedicated/reference/sql/functions/misc/#interpolate),
+[locf](/influxdb/cloud-dedicated/reference/sql/functions/misc/#locf)
+
+{{< expand-wrapper >}}
+{{% expand "View `date_bin_wallclock_gapfill` query examples" %}}
```

Comment:
Is this better than individual expands for each example?

Reply:
I don't know that it's better. I copied this structure from `date_bin_gapfill`, but I think I might convert it into multiple expand blocks.